modules/core/include/opencv2/core/types.hpp

```diff
@@ -1215,7 +1215,7 @@ _Tp Point_<_Tp>::dot(const Point_& pt) const
 template<typename _Tp> inline
 double Point_<_Tp>::ddot(const Point_& pt) const
 {
-    return (double)x*pt.x + (double)y*pt.y;
+    return (double)x*(double)(pt.x) + (double)y*(double)(pt.y);
```

Comment:
Dear @bduclaux 

Although the patch can fix the issue, the casting can achieve with `static_cast`. 

Here, it's the code. 
```
return static_cast<double>(x)*static_cast<double>(pt.x) + 
  static_cast<double>(y)*static_cast<double>(pt.y);
```

I think it's better. Do you agree ?

Reply:
Yes, static_cast is indeed much better. 
Btw, there are tons of C-style cast in the code, and it will take months to fully check / rewrite them. Not a problem, unless the compiler emits warnings like above !
Thanks !